manticore/platforms/platform.py

```diff
@@ -21,12 +21,28 @@ def __init__(self, reg_num, message='Concretizing syscall argument', policy='SAM
         super(ConcretizeSyscallArgument, self).__init__(message)
 
 
-class Platform(object):
+class Platform(Eventful):
     '''
     Base class for all operating system platforms.
     '''
-    def __init__(self, path):
-        self._path = path
+    def __init__(self, path, **kwargs):
+        super(Platform, self).__init__(**kwargs)
+        self._path = path #Not clear why all platforms must have a "path"
```

Comment:
So that program must always be on a file? Can't we  make it in the air?
Like for analyzing a bunch of code/shellcode.